src/tools/run-make-support/src/command.rs

```diff
@@ -137,11 +179,11 @@ impl Command {
     fn command_output(&mut self) -> CompletedProcess {
         self.drop_bomb.defuse();
         // let's make sure we piped all the input and outputs
-        self.cmd.stdin(Stdio::piped());
-        self.cmd.stdout(Stdio::piped());
-        self.cmd.stderr(Stdio::piped());
+        self.cmd.stdin(self.stdin.take().unwrap_or(Stdio::piped()));
+        self.cmd.stdout(self.stdout.take().unwrap_or(Stdio::piped()));
+        self.cmd.stderr(self.stderr.take().unwrap_or(Stdio::piped()));
```

Comment:
I think this isn't quite correct because `command_output` does not take ownership of `self`, meaning that if you call `run()` and thus `command_output` twice this will be different.

Reply:
Does it? I assume that almost everywhere we just build the command, use `run_xxx` and then work with `CompletedProcess`, or not?

Reply:
I'll tinker with it a bit more later.

Reply:
Right, this is always annoying. Shame that `Stdio` does not support `Clone`. Well, we could make all the methods take `self`. That would probably be too annoying, but I'd try it nevertheless, just to see the fallout - maybe we'll see that in most tests it's actually fine.

Another option would be to add a special `run` method that takes the Stdio wrappers, rather than storing them inside the command. Or just do the `take().unwrap()` thing - executing the same command twice should be an error anyway, we could explicitly assert it.

Reply:
Yeah I think we might just assert the same command should not be `run()` twice, since neither our `Command` nor `std`'s `Command` can be cloned anyway.

Reply:
> we could make all the methods take `self`.

I think this is a no-go from the start due to need for conditional args, like

```rs
let mut cmd = rustc();

if is_today_thursday() {
    cmd.arg("--explode");
}

cmd.run();
```

so we can't have the intermediate helpers take ownership.

Reply:
I mean, it's not that difficult, you can just use `cmd = cmd.arg("--explode");`, it's a common pattern in builders. But I don't think it's worth it here, let's just assert that double run is forbidden.

Reply:
Changed to assert only run once.